kong/tools/ip.lua

```diff
@@ -0,0 +1,56 @@
+local ip = require "resty.mediador.ip"
+local px = require "resty.mediador.proxy"
+
+
+local function trust_all()  return true  end
+local function trust_none() return false end
+
+
+local _M = { valid = ip.valid }
+
+
+function _M.init(config)
+  local ips = config.trusted_ips
+  local count = #ips
+  local trusted_ips = {}
+  local trust_all_ipv4, trust_all_ipv6
+
+  -- This is because we don't support unix: that the ngx_http_realip module supports.
+  -- Also as an optimization we will only compile trusted ips if the Kong is not run
+  -- with the default 0.0.0.0/0, ::/0 aka trust all ip addresses settings.
+  for i = 1, count do
+    local address = ips[i]
+
+    if ip.valid(address) then
+      trusted_ips[#trusted_ips+1] = address
+      if address == "0.0.0.0/0" then
+        trust_all_ipv4 = true
+
+      elseif address == "::/0" then
+        trust_all_ipv6 = true
+      end
+    end
+  end
+
+  if #trusted_ips == 0 then
+    return {
+      valid   = ip.valid,
+      trusted = trust_none,
+    }
+  end
+
+  if trust_all_ipv4 and trust_all_ipv6 then
```

Comment:
Yeah sorry I wasn't very clear here. Yes, `--with-ipv6` was dropped. What I meant is split in 2 parts:
- I *think* that users should be able to enable/disable ipv6 support through Kong
- Currently doable via the `--with-nginx` option check in the current OpenResty (I think `1.11.2`still has it? not sure?), but in the future, the need for disabling ipv6 will be even greater, because the OS might support ipv6, but not the network, and we cannot rely on the `--with-ipv6` option anymore to assume what the user wanted, hence, it simply increased the need for this behavior to be configurable.